IPython/core/ultratb.py

```diff
@@ -1094,12 +1094,14 @@ def structured_traceback(self, etype, evalue, etb, tb_offset=None,
                 etype, evalue, etb = exception
             else:
                 evalue = None
+            chained_exc_ids = set()
```

Comment:
I've let it like that intentionnaly this way to allow seeing there is a cycle.. If you change it your way you should then add a message of loop detection... Helping users to detect a potential issue...